### ISSUE TITLE
Rule quoted-strings enforce quote-type unless the string contains quotes or escape sequences

### ISSUE THREAD

**Issue by bendem** [adrienverge/yamllint#424]
Hello, `quoted-strings` with `quote-type` is a great rule for uniformity, but there are times when it just doesn't make sense. Once of those times is when we use quotes inside the string. In that case, it doesn't make sense to enforce the quote-type.
```yaml
something: 'use the "force" luke'  # yamllint tells me I should write "use the \"force\" luke"
```

Another case is when `quote-type` is set to `double`, but the string contains escape sequences. In those cases, single quote is better, since the content is not interpreted (so we don't need to double backslashes):
```yaml
regexp: '^(.*\scifs\s.*)vers=[12].0(.*)$'  # yamllint tells me I should write regexp: "^(.*\\scifs\\s.*)vers=[12].0(.*)$"
```

**Comment by adrienverge** [comment]
Hello, interesting issue! I guess you've already looked [`required: only-when-needed`](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.quoted_strings) and it doesn't help here.

Do you have an idea for a new option (or extending an option) so that yamllint can handle that?

**Comment by mwgamble** [comment]
I've just run into this while linting some Cloudformation YAML that contains DNS record configuration. How about `prefer-double` and `prefer-single`, with the caveat that the style that results in the least amount of escaping is best?

**Comment by mwgamble** [comment]
I see there's a partial solution implemented in #487 and merged. Is there any timeline for releasing this change?

**Comment by bendem** [comment]
> I see there's a partial solution implemented in #487 and merged. Is there any timeline for releasing this change?

I totally missed the PR, this is great.

As expected, escape exception is harder, but hey, it's a first step 🎉

**Comment by adrienverge** [comment]
> Is there any timeline for releasing this change?

I just released version 1.28.0 for this :+1: 

I'll close the issue, because https://github.com/adrienverge/yamllint/pull/487 aims to solve this problem, and adding new options like `prefer-single` or `prefer-double` would be redundant and conflict with `allow-quoted-quotes`.

TL;DR: users facing [the problem in the example above ↑](https://github.com/adrienverge/yamllint/issues/424#issue-1053447948) can use:
```yaml
rules:
  quoted-strings:
    quote-type: single
    allow-quoted-quotes: true
```
More details: https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.quoted_strings

**Comment by mwgamble** [comment]
Thank you so much! :heart:
